pkg/version/version.go

```diff
@@ -14,18 +14,94 @@
  * limitations under the License.
  */
 
+// Reference: https://semver.org; NOT strictly followed.
 package version
 
-import "fmt"
-
-var (
-	version      = "1.0-dev"
-	revision     = "$Format:%h$"
-	revisionDate = "$Format:%as$"
+import (
+	"fmt"
+	"strconv"
+	"strings"
 )
 
-// Version returns version in format - `VERSION (REVISIONDATE REVISION)`
-// value is assigned in Makefile
+var ver = Semver{
+	major:        1,
+	minor:        0,
+	patch:        0,
+	preRelease:   "dev",
+	revision:     "$Format:%h$", // value is assigned in Makefile
+	revisionDate: "$Format:%as$",
+}
+
+type Semver struct {
+	major, minor, patch uint64
+	preRelease          string
+	revision            string
+	revisionDate        string
+}
+
 func Version() string {
-	return fmt.Sprintf("%v (%v %v)", version, revisionDate, revision)
+	pr := ver.preRelease
+	if pr != "" {
+		pr = "-" + pr
+	}
+	return fmt.Sprintf("%d.%d.%d%s+%s.%s",
+		ver.major, ver.minor, ver.patch, pr, ver.revisionDate, ver.revision)
```

Comment:
ver.revisionDate could be invalid, just ignore it

Reply:
will be set as "unknown"